extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/NarayanaQuarkusConfiguration.java

```diff
@@ -0,0 +1,147 @@
+    public List<String> expiryScannerClassNames;
+    /**
+     * config
+     */
+    @ConfigItem(name = "RecoveryEnvironmentBean.recoveryPort", defaultValue = "4712")
+    public int recoveryPort;
+    /**
+     * config
+     */
+    @ConfigItem(name = "RecoveryEnvironmentBean.recoveryAddress")
+    public String recoveryAddress;
+    /**
+     * config
+     */
+    @ConfigItem(name = "RecoveryEnvironmentBean.transactionStatusManagerPort", defaultValue = "0")
+    public int transactionStatusManagerPort;
+    /**
+     * config
+     */
+    @ConfigItem(name = "RecoveryEnvironmentBean.transactionStatusManagerAddress")
+    public String transactionStatusManagerAddress;
+    /**
+     * config
+     */
+    @ConfigItem(name = "RecoveryEnvironmentBean.recoveryListener", defaultValue = "true")
+    public boolean recoveryListener;
+
+    // non final properties
+    /**
+     * config
+     */
+    @ConfigItem(name = "CoordinatorEnvironmentBean.readonlyOptimisation", defaultValue = "true")
+    public boolean readonlyOptimisation;
+    /**
+     * config
+     */
+    @ConfigItem(name = "CoordinatorEnvironmentBean.startDisabled", defaultValue = "false")
+    public boolean startDisabled;
+    /**
+     * config
+     */
+    @ConfigItem(name = "CoordinatorEnvironmentBean.defaultTimeout", defaultValue = "60")
+    public int defaultTimeout;
+    /**
+     * config
+     */
+    @ConfigItem(name = "CoordinatorEnvironmentBean.beforeCompletionWhenRollbackOnly", defaultValue = "false")
+    public boolean beforeCompletionWhenRollbackOnly;
+
+    // final properties
+    /**
+     * config
+     */
+    @ConfigItem(name = "CoordinatorEnvironmentBean.maintainHeuristics", defaultValue = "true")
+    public boolean maintainHeuristics;
+    /**
+     * config
+     */
+    @ConfigItem(name = "CoordinatorEnvironmentBean.asyncCommit", defaultValue = "false")
+    public boolean asyncCommit;
+    /**
+     * config
+     */
+    @ConfigItem(name = "CoordinatorEnvironmentBean.asyncPrepare", defaultValue = "false")
+    public boolean asyncPrepare;
+    /**
+     * config
+     */
+    @ConfigItem(name = "CoordinatorEnvironmentBean.asyncRollback", defaultValue = "false")
+    public boolean asyncRollback;
+    /**
+     * config
+     */
+    @ConfigItem(name = "CoordinatorEnvironmentBean.asyncBeforeSynchronization", defaultValue = "false")
+    public boolean asyncBeforeSynchronization;
+    /**
+     * config
+     */
+    @ConfigItem(name = "CoordinatorEnvironmentBean.asyncAfterSynchronization", defaultValue = "false")
+    public boolean asyncAfterSynchronization;
+    /**
+     * config
+     */
+    @ConfigItem(name = "CoordinatorEnvironmentBean.dynamic1PC", defaultValue = "true")
+    public boolean dynamic1PC;
+
+    //    @ConfigItem(name = "CoordinatorEnvironmentBean.transactionStatusManagerEnable", defaultValue = "false")
```

Comment:
I assume there should not be a commented out text in the code. If it's not necessary it should be just removed.

Reply:
Yes I was waiting for the review to happen before deciding on which properties we want to be configurable and which ones we want setting at build time. The initial PR just takes all of the existing defaults and makes them configurable. It is a basis for us to decide which should be set at build time and which ones we want to be configurable.